language/testing-infra/functional-tests/src/evaluator.rs

```diff
@@ -170,6 +179,13 @@ impl EvaluationLog {
         res
     }
 
+    pub fn to_text_for_matching(&self) -> Vec<String> {
```

Comment:
I think you are going to run into a lot of issues if you let the log define how it gets printed. I think this could be simpler if you extract all of this logic out and just put it into `check_or_update_expected_output`

Reply:
In general I totally agree with you and to be clear I've already run into some of these issues. I think the real problem here is that the checker based flow requires a log with different properties than the exp file based one. Trying to make them work together is a losing battle to start with. I'd be happy to put in some minimum effort to make it usable as I've already done, but would prefer avoiding going deeper in this rabbit hole and focus on building the new stuff. The new one will simply have a stream you can `write!` into (kinda like the stdout).